TEZ-4129: Delete intermediate attempt data for failed attempts for Shuffle Handler

shameersss1 opened the pull request:
Delete intermediate attempt data for failed task attempts for Tez Shuffle Handler

abstractdog commented:
@shameersss1: could you please rebase this PR and squash your commits? I believe ShuffleHandler changes should be reviewed and committed next year, we'll find a way to review those
I think Github PRs are going to make the conversation much easier

shameersss1 commented:
> @shameersss1: could you please rebase this PR and squash your commits? I believe ShuffleHandler changes should be reviewed and committed next year, we'll find a way to review those I think Github PRs are going to make the conversation much easier

@abstractdog Thanks for looking into this. I have rebased and squash merged the commit. Please review the changes.

tez-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  16m 52s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 3 new or modified test files.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   4m 33s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   8m 59s |  master passed  |
| +1 :green_heart: |  compile  |   2m 49s |  master passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  compile  |   2m 41s |  master passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  checkstyle  |   2m 36s |  master passed  |
| +1 :green_heart: |  javadoc  |   2m 48s |  master passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  javadoc  |   2m 30s |  master passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +0 :ok: |  spotbugs  |   0m 43s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m 52s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  Maven dependency ordering for patch  |
| -1 :x: |  mvninstall  |   0m 14s |  tez-aux-services in the patch failed.  |
| -1 :x: |  compile  |   0m 16s |  tez-aux-services in the patch failed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04.  |
| -1 :x: |  javac  |   0m 16s |  tez-aux-services in the patch failed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04.  |
| -1 :x: |  compile  |   0m 14s |  tez-aux-services in the patch failed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07.  |
| -1 :x: |  javac  |   0m 14s |  tez-aux-services in the patch failed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07.  |
| -0 :warning: |  checkstyle  |   0m 11s |  tez-plugins/tez-aux-services: The patch generated 2 new + 63 unchanged - 0 fixed = 65 total (was 63)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   1m 29s |  the patch passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  javadoc  |   1m 23s |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| -1 :x: |  findbugs  |   0m 16s |  tez-aux-services in the patch failed.  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   1m 58s |  tez-api in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 27s |  tez-common in the patch passed.  |
| +1 :green_heart: |  unit  |   5m 12s |  tez-runtime-library in the patch passed.  |
| +1 :green_heart: |  unit  |   4m 16s |  tez-dag in the patch passed.  |
| -1 :x: |  unit  |   0m 14s |  tez-aux-services in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 47s |  The patch does not generate ASF License warnings.  |
|  |   |  76m 12s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-72/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/72 |
| JIRA Issue | TEZ-4129 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs checkstyle compile |
| uname | Linux f1d176f0b9e6 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / c9b8e90db |
| Default Java | Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
| mvninstall | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-72/1/artifact/out/patch-mvninstall-tez-plugins_tez-aux-services.txt |
| compile | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-72/1/artifact/out/patch-compile-tez-plugins_tez-aux-services-jdkUbuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04.txt |
| javac | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-72/1/artifact/out/patch-compile-tez-plugins_tez-aux-services-jdkUbuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04.txt |
| compile | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-72/1/artifact/out/patch-compile-tez-plugins_tez-aux-services-jdkPrivateBuild-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07.txt |
| javac | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-72/1/artifact/out/patch-compile-tez-plugins_tez-aux-services-jdkPrivateBuild-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07.txt |
| checkstyle | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-72/1/artifact/out/diff-checkstyle-tez-plugins_tez-aux-services.txt |
| findbugs | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-72/1/artifact/out/patch-findbugs-tez-plugins_tez-aux-services.txt |
| unit | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-72/1/artifact/out/patch-unit-tez-plugins_tez-aux-services.txt |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-72/1/testReport/ |
| Max. process+thread count | 1099 (vs. ulimit of 5500) |
| modules | C: tez-api tez-common tez-runtime-library tez-dag tez-plugins/tez-aux-services U: . |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-72/1/console |
| versions | git=2.25.1 maven=3.6.3 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

tez-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 29s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 3 new or modified test files.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   4m 22s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   9m  8s |  master passed  |
| +1 :green_heart: |  compile  |   2m 50s |  master passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  compile  |   2m 39s |  master passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  checkstyle  |   2m 34s |  master passed  |
| +1 :green_heart: |  javadoc  |   2m 48s |  master passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  javadoc  |   2m 31s |  master passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +0 :ok: |  spotbugs  |   0m 42s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m 49s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  9s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 45s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 40s |  the patch passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  javac  |   1m 40s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 30s |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  javac  |   1m 30s |  the patch passed  |
| -0 :warning: |  checkstyle  |   0m 11s |  tez-plugins/tez-aux-services: The patch generated 2 new + 63 unchanged - 0 fixed = 65 total (was 63)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   1m 29s |  the patch passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  javadoc  |   1m 23s |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  findbugs  |   4m  5s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m  0s |  tez-api in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 27s |  tez-common in the patch passed.  |
| +1 :green_heart: |  unit  |   5m 19s |  tez-runtime-library in the patch passed.  |
| +1 :green_heart: |  unit  |   4m 16s |  tez-dag in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 38s |  tez-aux-services in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 48s |  The patch does not generate ASF License warnings.  |
|  |   |  62m 51s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-72/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/72 |
| JIRA Issue | TEZ-4129 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs checkstyle compile |
| uname | Linux a5b2e48fce9c 4.15.0-156-generic #163-Ubuntu SMP Thu Aug 19 23:31:58 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / c9b8e90db |
| Default Java | Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
| checkstyle | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-72/2/artifact/out/diff-checkstyle-tez-plugins_tez-aux-services.txt |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-72/2/testReport/ |
| Max. process+thread count | 1479 (vs. ulimit of 5500) |
| modules | C: tez-api tez-common tez-runtime-library tez-dag tez-plugins/tez-aux-services U: . |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-72/2/console |
| versions | git=2.25.1 maven=3.6.3 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

abstractdog commented on tez-plugins/tez-aux-services/src/main/java/org/apache/tez/auxservices/ShuffleHandler.java:
```diff
@@ -1421,14 +1462,14 @@ protected ChannelFuture sendMapOutput(ChannelHandlerContext ctx, Channel ch,
       ChannelFuture writeFuture;
       if (ch.pipeline().get(SslHandler.class) == null) {
-        final FadvisedFileRegion partition = new FadvisedFileRegion(spill,
+        final org.apache.tez.auxservices.FadvisedFileRegion partition = new org.apache.tez.auxservices.FadvisedFileRegion(spill,
```
I don't think a fully qualified class name is needed here, as FadvisedFileRegion is in the same package

shameersss1 replied:
ack

abstractdog commented on tez-plugins/tez-aux-services/src/main/java/org/apache/tez/auxservices/ShuffleHandler.java:
```diff
@@ -1421,14 +1462,14 @@ protected ChannelFuture sendMapOutput(ChannelHandlerContext ctx, Channel ch,
       } else {
         // HTTPS cannot be done with zero copy.
-        final FadvisedChunkedFile chunk = new FadvisedChunkedFile(spill,
+        final org.apache.tez.auxservices.FadvisedChunkedFile chunk = new org.apache.tez.auxservices.FadvisedChunkedFile(spill,
```
same here as above

shameersss1 replied:
ack

abstractdog commented on tez-api/src/main/java/org/apache/tez/dag/api/TezConfiguration.java:
```diff
@@ -902,6 +902,12 @@ public TezConfiguration(boolean loadDefaults) {
           TEZ_AM_PREFIX + "dag.appcontext.thread-count-limit";
 
   public static final int TEZ_AM_DAG_APPCONTEXT_THREAD_COUNT_LIMIT_DEFAULT = 10;
+  /** Boolean value. Instructs AM to delete intermediate attempt data for failed task attempts */
```
1. nit: leave 1 line space between conf properties
2. put this right after TEZ_AM_DAG_CLEANUP_THREAD_COUNT_LIMIT in order to get cleanup related properties close to each other
3. there is a comment on TEZ_AM_DAG_CLEANUP_THREAD_COUNT_LIMIT:
```
   * Int value. Upper limit on the number of threads used to delete DAG directories on nodes.
```
as far as I can see this patch doesn't introduce a new executor for cleaning up failed attempt data, instead, it uses DeletionTrackerImpl.dagCleanupService, so this comment should be changed to reflect that this thread count is shared between dag cleanup + failed attempt cleanup

shameersss1 replied:
ack

abstractdog commented on tez-plugins/tez-aux-services/src/main/java/org/apache/tez/auxservices/ShuffleHandler.java:
```diff
@@ -1136,6 +1142,41 @@ private boolean deleteDagDirectories(Channel channel,
+      if (jobQ == null || jobQ.isEmpty()) {
+        return false;
+      }
+      if (taskAttemptFailedQ != null && !taskAttemptFailedQ.isEmpty() && taskAttemptFailedQ.get(0).contains("delete")
```
I can see this boilerplate in more places in shufflehandler, which is because of the nature of QueryStringDecoder.get(...) -> List<String>
could you please refactor this into common methods, like:
```
notEmpty(taskAttemptFailedQ)
notEmptyAndContains(taskAttemptFailedQ, "delete")
```

shameersss1 replied:
ack

shameersss1 commented on tez-dag/src/test/java/org/apache/tez/dag/app/dag/impl/TestTaskImpl.java:
```diff
@@ -394,7 +394,7 @@ public void testKillRunningTask() {
 
   @Test(timeout = 5000)
-  public void testTooManyFailedAttempts() {
+  public void testTooManyFailedAtetmpts() {
```
ack

shameersss1 commented:
@abstractdog  - Thanks for the review. I have addressed your comments. Please re-review the changes

tez-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  3s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 2 new or modified test files.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   5m  4s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   9m 21s |  master passed  |
| +1 :green_heart: |  compile  |   2m 52s |  master passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  compile  |   2m 41s |  master passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  checkstyle  |   2m 36s |  master passed  |
| +1 :green_heart: |  javadoc  |   2m 49s |  master passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  javadoc  |   2m 31s |  master passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +0 :ok: |  spotbugs  |   0m 43s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m 54s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 46s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 43s |  the patch passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  javac  |   1m 43s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 34s |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  javac  |   1m 34s |  the patch passed  |
| -0 :warning: |  checkstyle  |   0m 10s |  tez-plugins/tez-aux-services: The patch generated 2 new + 63 unchanged - 0 fixed = 65 total (was 63)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   1m 31s |  the patch passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  javadoc  |   1m 24s |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  findbugs  |   4m  5s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m  6s |  tez-api in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 28s |  tez-common in the patch passed.  |
| +1 :green_heart: |  unit  |   4m 34s |  tez-runtime-library in the patch passed.  |
| +1 :green_heart: |  unit  |   4m 25s |  tez-dag in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 43s |  tez-aux-services in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 48s |  The patch does not generate ASF License warnings.  |
|  |   |  64m 19s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-72/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/72 |
| JIRA Issue | TEZ-4129 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs checkstyle compile |
| uname | Linux e77a7a965b00 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / 1176386c9 |
| Default Java | Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
| checkstyle | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-72/3/artifact/out/diff-checkstyle-tez-plugins_tez-aux-services.txt |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-72/3/testReport/ |
| Max. process+thread count | 1485 (vs. ulimit of 5500) |
| modules | C: tez-api tez-common tez-runtime-library tez-dag tez-plugins/tez-aux-services U: . |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-72/3/console |
| versions | git=2.25.1 maven=3.6.3 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

abstractdog commented on tez-plugins/tez-aux-services/src/test/java/org/apache/tez/auxservices/TestShuffleHandler.java:
```diff
@@ -1312,6 +1312,87 @@ protected void sendError(ChannelHandlerContext ctx, String message,
+                  ShuffleHandler.USERCACHE, user,
+                  ShuffleHandler.APPCACHE, appId.toString(), "dag_1/output/", appAttemptId});
+      File taskAttemptDir = new File(taskAttemptDirStr);
+      Assert.assertTrue("Task Attempt Directory does not exist!", taskAttemptDir.exists());
```
for clarity's sake, please create an Assert.assertFalse(..., taskAttemptDir.exists()) before you expect the file to appear, this way the unit test also verifies that the task attempt file wasn't present (it's not the point of the unit test, but it worth doing)

shameersss1 replied:
ack!

abstractdog commented:
> @abstractdog - Thanks for the review. I have addressed your comments. Please re-review the changes

thanks @shameersss1 , 1 more very minor note

shameersss1 commented:
> > @abstractdog - Thanks for the review. I have addressed your comments. Please re-review the changes
> 
> thanks @shameersss1 , 1 more very minor note

@abstractdog I have addressed that. Please re-review the changes

tez-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 10s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 2 new or modified test files.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   4m 59s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   9m 18s |  master passed  |
| +1 :green_heart: |  compile  |   2m 52s |  master passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  compile  |   2m 39s |  master passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  checkstyle  |   2m 36s |  master passed  |
| +1 :green_heart: |  javadoc  |   2m 48s |  master passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  javadoc  |   2m 34s |  master passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +0 :ok: |  spotbugs  |   0m 42s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m 55s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 48s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 43s |  the patch passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  javac  |   1m 43s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 32s |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  javac  |   1m 32s |  the patch passed  |
| -0 :warning: |  checkstyle  |   0m 11s |  tez-plugins/tez-aux-services: The patch generated 4 new + 63 unchanged - 0 fixed = 67 total (was 63)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   1m 29s |  the patch passed with JDK Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04  |
| +1 :green_heart: |  javadoc  |   1m 23s |  the patch passed with JDK Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  findbugs  |   4m  7s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m  4s |  tez-api in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 27s |  tez-common in the patch passed.  |
| +1 :green_heart: |  unit  |   4m 35s |  tez-runtime-library in the patch passed.  |
| +1 :green_heart: |  unit  |   4m 26s |  tez-dag in the patch passed.  |
| +1 :green_heart: |  unit  |   2m 44s |  tez-aux-services in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 49s |  The patch does not generate ASF License warnings.  |
|  |   |  64m 20s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-72/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/72 |
| JIRA Issue | TEZ-4129 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs checkstyle compile |
| uname | Linux ccdbac750454 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / 1176386c9 |
| Default Java | Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.13+8-Ubuntu-0ubuntu1.20.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_312-8u312-b07-0ubuntu1~20.04-b07 |
| checkstyle | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-72/4/artifact/out/diff-checkstyle-tez-plugins_tez-aux-services.txt |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-72/4/testReport/ |
| Max. process+thread count | 2100 (vs. ulimit of 5500) |
| modules | C: tez-api tez-common tez-runtime-library tez-dag tez-plugins/tez-aux-services U: . |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-72/4/console |
| versions | git=2.25.1 maven=3.6.3 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

abstractdog reviewed (approved):
LGTM +1
I've included this patch to a cluster using tez shuffle handler, haven't found any regression
code looks straightforward and clean, I'm approving

letting this PR open for 24h if any other comments come in
@jteagles
